contrib_bots/lib/john.py

```diff
@@ -0,0 +1,122 @@
+# pip install chatterbot is necessary
+import json
+import os
+import time
+
+from random import choice
+
+from chatterbot import ChatBot
+from chatterbot.trainers import ChatterBotCorpusTrainer, ListTrainer
+
+JOKES_PATH = os.path.expanduser('lib/John/jokes.json')
+DATABASE_PATH = os.path.expanduser('~/JohnBot/database.db')
+DIRECTORY_PATH = os.path.expanduser('~/JohnBot')
+
+if not os.path.exists(DIRECTORY_PATH):
+    os.makedirs(DIRECTORY_PATH)
+
+# Create a new instance of a ChatBot
+def CreateChatBot(nolearn):
+    return ChatBot("John",
+                   storage_adapter="chatterbot.storage.JsonFileStorageAdapter",
+                   logic_adapters=
+                   [
+                      "chatterbot.logic.MathematicalEvaluation",
+                      {
+                          "import_path": "chatterbot.logic.BestMatch",
+                          "response_selection_method": "chatterbot.response_selection.get_random_response",
+                          "statement_comparison_function": "chatterbot.comparisons.levenshtein_distance"
+                      }],
+                   output_adapter="chatterbot.output.OutputFormatAdapter",
+                   output_format='text',
+                   database=DATABASE_PATH,
+                   silence_performance_warning="True",
+                   read_only=nolearn)
+
+bot = CreateChatBot("False")
+bot.set_trainer(ListTrainer)
+
+bot.train([
+    "I want to contribute",
+    """Contributors are more than welcomed! Please read
+    https://github.com/zulip/zulip#how-to-get-involved-with-contributing-to-zulip
+    to learn how to contribute.""",
+])
+
+bot.train([
+    "What is Zulip?",
+    """Zulip is a powerful, open source group chat application. Written in Python
+    and using the Django framework, Zulip supports both private messaging and group
+    chats via conversation streams. You can learn more about the product and its
+    features at https://www.zulip.org.""",
+])
+
+bot.train([
+    "I would like to request a remote dev instance",
+    """Greetings! You should receive a response from one of our mentors soon.
+    In the meantime, why don't you learn more about running Zulip on a development
+    environment? https://zulip.readthedocs.io/en/latest/using-dev-environment.html""",
+])
+
+bot.train([
+    "Joke!",
+    "Only if you ask nicely!",
+])
+
+bot.train([
+    "What is your name?",
+    "I am John, my job is to assist you with Zulip.",
+])
+
+bot.train([
+    "What can you do?",
+    "I can provide useful information and jokes if you follow etiquette.",
+])
+
+with open(JOKES_PATH) as data_file:
+    for joke in json.load(data_file):
+        bot.train([
+            "Please can you tell me a joke?",
+            joke['joke'],
+        ])
+
+bot.set_trainer(ChatterBotCorpusTrainer)
+
+bot.train(
+    "chatterbot.corpus.english"
+)
+
+bot = CreateChatBot("True")
```

Comment:
@TomaszKolek We need this to be false at the beginning so that the bot is able to learn. Then we set this to true, so that when we deploy the bot to Zulip it is unable to learn. If a malicious user tries to teach the bot insults, he won't be able to. More info here: http://chatterbot.readthedocs.io/en/stable/quickstart.html?highlight=read_only